src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -549,7 +549,31 @@ private function formatControllerLink($controller, string $anchorText): string
                 $r = new \ReflectionFunction($controller);
             }
         } catch (\ReflectionException $e) {
-            return $anchorText;
+            $id = $controller;
+            $method = '__invoke';
+
+            if ($pos = strpos($controller, '::')) {
+                $id = substr($controller, 0, $pos);
+                $method = substr($controller, $pos + 2);
+            }
+
+            if (!$getContainer || !($container = $getContainer()) || !$container->has($id)) {
+                return $anchorText;
+            }
+
+            try {
+                $definition = $container->findDefinition($id);
+                $factory = $definition->getFactory();
+                if (($factory[1] ?? null) && !$factory[0] instanceof Definition) {
+                    $r = new \ReflectionMethod($factory[0], $factory[1]);
```

Comment:
I think this logic is wrong: we don't care to link to the factory; we want to link to the class that has the action method. It's the class that is *returned* by the factory, right? Since we require the definition to always have a class, this is the one, isn't it?

Reply:
Right, updated